Clang 19.1.7 doesn't compile current master

**Steps to reproduce:**
1. clone engine with the submodules
2. `cmake -G Ninja -B build -DCMAKE_BUILD_TYPE=Debug -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_C_COMPILER=clang -DINSTALL_PORTABLE=ON -DAI_TYPES=NATIVE`

**Error:**
```
In file included from /home/technate/Workspace/spring/rts/Game/Game.cpp:3:
In file included from /home/technate/Workspace/spring/rts/Rendering/GL/myGL.h:14:
In file included from /home/technate/Workspace/spring/rts/System/float3.h:13:
In file included from /home/technate/Workspace/spring/rts/System/creg/creg_cond.h:17:
In file included from /home/technate/Workspace/spring/rts/System/creg/creg.h:11:
In file included from /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/14.2.1/../../../../include/c++/14.2.1/memory:78:
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/14.2.1/../../../../include/c++/14.2.1/bits/unique_ptr.h:91:16: error: invalid application of 'sizeof' to an incomplete type 'IWater'
   91 |         static_assert(sizeof(_Tp)>0,
      |                       ^~~~~~~~~~~
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/14.2.1/../../../../include/c++/14.2.1/bits/unique_ptr.h:399:4: note: in instantiation of member function 'std::default_delete<IWater>::operator()' requested here
  399 |           get_deleter()(std::move(__ptr));
      |           ^
/home/technate/Workspace/spring/rts/Rendering/Env/IWater.h:55:48: note: in instantiation of member function 'std::unique_ptr<IWater>::~unique_ptr' requested here
   55 |         static inline std::unique_ptr<IWater> water = nullptr;
      |                                                       ^
/home/technate/Workspace/spring/rts/Rendering/Env/IWater.h:12:7: note: definition of 'IWater' is not complete until the closing '}'
   12 | class IWater : public IExplosionListener
```

**Fix:**
```
-set(CMAKE_CXX_STANDARD 23)
+set(CMAKE_CXX_STANDARD 20)
```

**My environment:**
Arch Linux

```
clang version 19.1.7
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-pc-linux-gnu/14.2.1
Found candidate GCC installation: /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/14.2.1
Selected GCC installation: /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/14.2.1
Candidate multilib: .;@m64
Candidate multilib: 32;@m32
Selected multilib: .;@m64
``` 

It seems that change to `-std=c++23` brings incoherence between compiler implementations.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,7 +2,7 @@ cmake_minimum_required(VERSION 3.27)
 cmake_policy(SET CMP0060 OLD) #causes link errors on Linux otherwise
 cmake_policy(SET CMP0065 OLD) #we need binaries compiled with -rdynamic for BARb to work (cause unknown https://github.com/beyond-all-reason/spring/issues/405)
 
-set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD 23)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 

